influxql/parser.go

```diff
@@ -864,14 +864,29 @@ func (p *Parser) parseShowFieldKeysStatement() (*ShowFieldKeysStatement, error)
 // This function assumes the "DROP SERIES" tokens have already been consumed.
 func (p *Parser) parseDropSeriesStatement() (*DropSeriesStatement, error) {
 	stmt := &DropSeriesStatement{}
+	var err error
 
-	// Read the name of the series to drop.
-	lit, err := p.parseIdent()
-	if err != nil {
-		return nil, err
-	}
-	stmt.Name = lit
+	tok, _, _ := p.scanIgnoreWhitespace()
+	if tok == FROM {
+		// Parse source.
+		if stmt.Source, err = p.parseSource(); err != nil {
+			return nil, err
+		}
 
+		// Parse condition: "WHERE EXPR".
+		if stmt.Condition, err = p.parseCondition(); err != nil {
+			return nil, err
+		}
+
+	} else {
+		p.unscan()
+		var id int
+		id, err = p.parseInt(0, math.MaxUint32)
+		if err != nil {
+			return nil, err
+		}
+		stmt.SeriesID = uint32(id)
+	}
```

Comment:
I should have thought of this earlier when we were discussing syntax but it needs to support the syntax I mentioned in the comment above, on the `String()` method.